web/api/join-callback/index.ts

```diff
@@ -75,7 +77,7 @@ export const POST = withApiAuthRequired(async (req: NextRequest) => {
   ironCladUserId = crypto.randomUUID();
 
   try {
-    const url = new URL(`${process.env.NEXT_PUBLIC_APP_URL}/signup`);
+    const url = new URL(`${appUrl}/signup`);
```

Comment:
```suggestion
    const url = new URL("/signup", appUrl);
```

Reply:
also maybe worth use it with `urls.signUp()` as it is in other places instead of "/signup"